MUXSDKKaltura/MUXSDKKalturaTests/MUXSDKStatsTests+Utilities.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  MUXSDKStatsTests+Utilities.swift
+//  MUXSDKKalturaTests
+//
+//  Created by Audra Rodriguez on 18/11/21.
+//  Copyright © 2021 Mux, Inc. All rights reserved.
+//
+
+import XCTest
+@testable import MUXSDKKaltura
+import MuxCore
+
+extension MUXSDKStatsTests {
+    enum EventType{
+        case customerVideoData
+        case customerPlayerData
+        case customerData
+        case customData
+    }
+    
+    func assertDispatchedEventTypesForPlayer(id: String, expectedEventTypes: [String]) {
+        let dispatchedEventsType = MockedDispatcher.shared.dispatchedEvents
+            .filter { $0.playerId == id }
+            .map { $0.event.getType() }
+        
+        XCTAssertEqual(expectedEventTypes, dispatchedEventsType)
+    }
+    
+    func assertDispatchedCustomerDataEventsAtIndex(
+        index: Int,
+        expectedCustomerVideoData: [String : Any]? = nil,
+        expectedCustomerPlayerData: [String : Any]? = nil,
+        expectedCustomerViewData: [String : Any]? = nil,
+        expectedCustomData: [String : Any]? = nil
+    )  {
+        let dispatchedEvent = MockedData.dispatcher.dispatchedEvents[index].event as? MUXSDKDataEvent
+        if let expectedCustomerVideoData = expectedCustomerVideoData {
+            assertCustomerData(type: .customerVideoData, dispatchedEvent: dispatchedEvent, expectedData: expectedCustomerVideoData)
+        }
+        if let expectedCustomerPlayerData = expectedCustomerPlayerData {
+            assertCustomerData(type: .customerPlayerData, dispatchedEvent: dispatchedEvent, expectedData: expectedCustomerPlayerData)
+        }
+        if let expectedCustomerViewData = expectedCustomerViewData {
+            assertCustomerData(type: .customerData, dispatchedEvent: dispatchedEvent, expectedData: expectedCustomerViewData)
+        }
+        if let expectedCustomData = expectedCustomData {
+            assertCustomerData(type: .customData, dispatchedEvent: dispatchedEvent, expectedData: expectedCustomData)
+        }
+    }
+    
+    func assertCustomerData(type: EventType, dispatchedEvent: MUXSDKDataEvent?, expectedData: [String : Any]) {
+        switch type {
+        case .customerVideoData:
+            guard let customerVideoData = dispatchedEvent?.customerVideoData else {
+                XCTAssertNil(dispatchedEvent?.customerVideoData)
```

Comment:
I'm not sure about this check. Cause if we got the type as customerVideoData it means that the expectedCustomerVideoData was different than nil. So in this case if the dispatchedEvent.customerVideoData is nil I think it would be more like a 
```
XCTFail("Customer video data for dispatched event not found")
```
But we might also want to check when we are not passing other data types to confirm that the dispatched event keeps those values as nil take the existing testSetCustomerData for example so maybe we create an assertNil that takes in the type and then in the `assertDispatchedCustomerDataEventsAtIndex` we check the assert nil when expectedDataType = nil.

Reply:
agreed